driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideEncryptionNotCreateMongocryptdClientTest.java

```diff
@@ -0,0 +1,224 @@
+    @Nullable
+    private static final String CRYPT_SHARED_LIB_PATH_SYS_PROP_VALUE = cryptSharedLibPathSysPropValue().orElse(null);
+    private static final int DEFAULT_MONGOCRYPTD_PORT = MongoCryptHelper.createMongocryptdClientSettings(null)
+            .getClusterSettings().getHosts().get(0).getPort();
+    private static final String LOCAL_KMS_PROVIDER_ID = "local";
+    private static final Duration TIMEOUT = Duration.ofMillis(1_000);
+    private static final MongoNamespace KEY_VAULT_NAMESPACE = new MongoNamespace("keyvault", "datakeys");
+
+    private MongoClient client;
+    private MongoCollection<Document> collection;
+    private ConnectionTracker mongocryptdConnectionTracker;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        assumeTrue(serverVersionAtLeast(4, 2));
+        assumeTrue(CRYPT_SHARED_LIB_PATH_SYS_PROP_VALUE != null);
+        mongocryptdConnectionTracker = ConnectionTracker.start();
+        client = createMongoClient(MongoClientSettings.builder(getMongoClientSettings())
+                .autoEncryptionSettings(AutoEncryptionSettings.builder()
+                        .kmsProviders(singletonMap(LOCAL_KMS_PROVIDER_ID, singletonMap("key", localKmsProviderKey())))
+                        .keyVaultNamespace(KEY_VAULT_NAMESPACE.getFullName())
+                        .extraOptions(Stream.of(
+                                new SimpleImmutableEntry<>("cryptSharedLibPath", CRYPT_SHARED_LIB_PATH_SYS_PROP_VALUE),
+                                new SimpleImmutableEntry<>("mongocryptdURI", format("mongodb://%s:%d/db?serverSelectionTimeoutMS=%d",
+                                        mongocryptdConnectionTracker.serverSocket().getInetAddress().getHostAddress(),
+                                        mongocryptdConnectionTracker.serverSocket().getLocalPort(),
+                                        TIMEOUT.toMillis()))
+                        ).collect(toMap(Entry::getKey, Entry::getValue)))
+                        .build())
+                .build());
+        client.getDatabase(KEY_VAULT_NAMESPACE.getDatabaseName()).drop();
+        MongoDatabase db = client.getDatabase("db");
+        db.drop();
+        collection = db.getCollection("coll");
+    }
+
+    @AfterEach
+    @SuppressWarnings("try")
+    public void cleanUp() throws Exception {
+        //noinspection unused
+        try (ConnectionTracker autoClosed = mongocryptdConnectionTracker;
+             MongoClient autoClosed2 = client) {
+            // empty
+        }
+    }
+
+    protected abstract MongoClient createMongoClient(MongoClientSettings settings);
+
+    @Test
+    @SuppressWarnings("try")
+    void whenCryptSharedLoaded() throws Exception {
+        //noinspection unused
+        try (AutoCloseable assertNoConnectionsOnAutoCloseToPreserveAssertionFailure = mongocryptdConnectionTracker) {
+            collection.insertOne(Document.parse("{unencrypted: 'test'}"));
+        }
```

Comment:
If we don't assert when automatically closing, then we may not see the assertion error if the driver fails behave correctly: if the driver attempts to spawn `mongocryptd`, it will fail as the socket is bound (by the test), which will lead to `insertOne` failing, which will result is skipping the assertion.

Reply:
Thanks for the explanation. One suggestion I have is to move the assertNoConnections() method to the finally block. I think it could make it easier to follow the context of the test (like in style of the given, when, then). If we move it to the finally block, it will be easier to locate the assertion quickly. What do you think?

```
        try (AutoCloseable assertNoConnectionsOnAutoCloseToPreserveAssertionFailure = mongocryptdConnectionTracker) {
                collection.insertOne(Document.parse("{unencrypted: 'test'}"));
        }finally {
            mongocryptdConnectionTracker.assertNoConnections();
        }
```

Reply:
Added a comment, as discussed.